src/core/Class.js

```diff
@@ -8,39 +8,68 @@ import * as Util from './Util';
 
 // Thanks to John Resig and Dean Edwards for inspiration!
 
-export function Class() {}
+export class Class {
+	// @function include(properties: Object): this
+	// [Includes a mixin](#class-includes) into the current class.
+	static include = function (props) {
+		const parentOptions = this.prototype.options;
+		Util.extend(this.prototype, props);
+		if (props.options) {
+			this.prototype.options = parentOptions;
+			this.mergeOptions(props.options);
+		}
+		return this;
+	};
+
+	// @function mergeOptions(options: Object): this
+	// [Merges `options`](#class-options) into the defaults of the class.
+	static mergeOptions(options) {
+		Util.extend(this.prototype.options, options);
+		return this;
+	}
+
+
+	// @function addInitHook(fn: Function): this
+	// Adds a [constructor hook](#class-constructor-hooks) to the class.
+	static addInitHook(fn, ...args) { // (Function) || (String, args...)
+		const init = typeof fn === 'function' ? fn : function () {
+			this[fn].apply(this, args);
+		};
+
+		this.prototype._initHooks = this.prototype._initHooks || [];
+		this.prototype._initHooks.push(init);
+		return this;
+	}
+}
 
 Class.extend = function (props) {
 
 	// @function extend(props: Object): Function
 	// [Extends the current class](#class-inheritance) given the properties to be included.
 	// Returns a Javascript function that is a class constructor (to be called with `new`).
-	const NewClass = function (...args) {
+	class NewClass {

```

Comment:
Try `class NewClass extends this.prototype { ... }` instead.

Reply:
Try also `const NewClass = class extends this.prototype` (a `class` *expression* instead of a `class` *statement*) - should make it an anonymous class and get rid of us seeing `NewClass` when debugging.

Reply:
I used the following code to extend from the prototype:
```js
const NewClass = class extends this.prototype {};
```
Unfortunately this produces the following exception:
```
Uncaught TypeError: Class extends value #<Class> is not a constructor or null
```

I believe that since this method is `static` we should actually be using `this` (without `prototype`):
```js
const NewClass = class extends this {};
```
And after adding the constructor it looks as follows:
```js
const NewClass = class extends this {
	constructor(...args) {
		super(...args);
		Util.setOptions(this);

		// call the constructor
		if (this.initialize) {
			this.initialize.apply(this, args);
		}

		// call all constructor hooks
		this.callInitHooks();
	}
};
```
After this change the class extension seems to work, however this starts to create errors when running the tests:
```
Error: Map container is already initialized.
	at NewClass._initContainer (src/LeafletWithGlobals.js:3733:12)
	at NewClass.initialize (src/LeafletWithGlobals.js:2779:10)
	at new NewClass (src/LeafletWithGlobals.js:243:24)
	at Object.createMap [as map] (src/LeafletWithGlobals.js:4397:11)
	at Context.<anonymous> (spec/suites/control/Control.AttributionSpec.js:6:11)
```

I think I'll have to move some of this logic such as the `constructor` and `callInitHooks()` to `Class` directly, I'll explore that a little.

Reply:
Managed to accomplish this in 3c65e75d90ae098aef7dff4b09609b2314b50ac9.